compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/CupertionTextFieldPointerModifier.skiko.kt

```diff
@@ -132,24 +133,86 @@ private fun getTapHandlerModifier(
     }
 }
 
+/**
+ * Returns a modifier which allows to precisely move the caret in the text by drag gesture after long press
+ *
+ * @param state The state of the text field.
+ * @param offsetMapping The offset mapping of the text field.
+ * @return A modifier that handles long press and drag gestures.
+ */
 @Composable
-private fun getSelectionModifier(manager: TextFieldSelectionManager): Modifier {
-    val currentManager by rememberUpdatedState(manager)
+private fun getLongPressHandlerModifier(
+    state: TextFieldState,
+    offsetMapping: OffsetMapping
+): Modifier {
+    val currentState by rememberUpdatedState(state)
+    val currentOffsetMapping by rememberUpdatedState(offsetMapping)
+    var dragTotalDistance = remember { Offset.Zero }
+    var dragBeginOffset = remember { Offset.Zero }
+    var currentDragPosition: Offset?
+
+    val longTapActionsObserver = object : TextDragObserver {
```

Comment:
```suggestion
    val longTapActionsObserver = remember {
        object : TextDragObserver {
    }
```

We shouldn't create a new object every recomposition and pass it to `pointerInput(Unit)` coroutine. We should create it once and `remember`.

If we won't do that, the coroutine will see the object only from the first recomposition. We will have a memory leak and input won't work, as it will manipulate the lost object.

Reply:
As an alternative to `remember`, we can move it to the coroutine itself:
```
return Modifier.pointerInput(Unit) {
    val longTapActionsObserver = object : TextDragObserver {
        ...
    }
    detectDragGesturesAfterLongPress(
```
This will look better

Reply:
Did I understand correctly that this in this particular case will have the similar effect like remember `TextDragObserver` object outside of the lambda's scope of `Modifier.pointerInput(Unit)`, but more lazy?
If there was more than one usage of `TextDragObserver` object, we would prefer to use remember instead?

Reply:
> similar effect like remember

Yes, the coroutine is only created once, and suspends on `detectDragGesturesAfterLongPress`

> but more lazy?

It is created at another time, yes, but this only give us a better code (related pieces defined in one place.)